vllm/model_executor/layers/layernorm.py

```diff
@@ -20,14 +20,19 @@ def __init__(
         hidden_size: int,
         eps: float = 1e-6,
         var_hidden_size: Optional[int] = None,
+        elementwise_affine: bool = True,
     ) -> None:
         super().__init__()
 
         self.hidden_size = hidden_size
         self.variance_epsilon = eps
         self.variance_size_override = (None if var_hidden_size == hidden_size
                                        else var_hidden_size)
-        self.weight = nn.Parameter(torch.ones(hidden_size))
+        self.elementwise_affine = elementwise_affine
+
+        self.weight = torch.ones(hidden_size)
+        if self.elementwise_affine:
+            self.weight = nn.Parameter(self.weight)
```

Comment:
Just curious: can we also gain performance improvement from a rms_norm kernel without weight multiply for `elementwise_affine=False`?

Reply:
My take on this is that the opportunity is pretty small, with RMSNorm typically taking maybe 2% of the time in the forward pass -- you would save on not having to load `self.weight` from GPU memory, but since it's just a vector this is _probably_ not a significant cost. And since we have custom layernorm kernels, modifying them to skip the weight application is probably not worth the effort.

_However_, I do think it's worth checking out if we can get speed up by skipping the weight application in the native forward method (using `torch.compile` for fusion)

Reply:
I tested the performance of weight application skipped in the native forward with `torch.compile`, and there is a minor performance speedup (about 4%) on long sequence compared to no skip. So I think we can skip the weight application in native forward, although the speedup is small. :)

![RMS_norm_performance](https://github.com/user-attachments/assets/7d9498ab-6835-4172-a79e-c52b3daafa19)
(Tested on a Tesla T4 16G)